custom_components/tahoma/config_flow.py

```diff
@@ -24,183 +17,37 @@
 )
 
 
-async def validate_input(hass: core.HomeAssistant, data):
-    """Validate the user input allows us to connect.
-
-    Data has the keys from DATA_SCHEMA with values provided by the user.
-    """
-    username = data.get(CONF_USERNAME)
-    password = data.get(CONF_PASSWORD)
-
-    try:
-        controller = await hass.async_add_executor_job(TahomaApi, username, password)
-
-    except RequestException:
-        _LOGGER.exception("Error when trying to log in to the TaHoma API")
-        raise CannotConnect
-
-    return_dict = {"title": username}
-    await hass.async_add_executor_job(controller.get_setup)
-    devices = await hass.async_add_executor_job(controller.get_devices)
-    for key, device in devices.items():
-        if device.uiclass == TAHOMA_TYPE_HEATING_SYSTEM:
-            if TAHOMA_TYPE_HEATING_SYSTEM not in return_dict:
-                return_dict[TAHOMA_TYPE_HEATING_SYSTEM] = {}
-            return_dict[TAHOMA_TYPE_HEATING_SYSTEM][device.url] = device.label
-
-    # If you cannot connect:
-    # throw CannotConnect
-    # If the authentication is wrong:
-    # InvalidAuth
-
-    # Return info that you want to store in the config entry.
-    return return_dict
-
-
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for TaHoma."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
-    @staticmethod
-    @callback
-    def async_get_options_flow(config_entry):
-        """Return the option flow."""
-        return ThermoOptionsFlowHandler(config_entry)
-
-    def __init__(self):
-        """Initialize the config flow."""
-        self._user_input = {}
-        self._thermos = {}
-
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         errors = {}
 
         if user_input is not None:
-            unique_id = user_input.get(CONF_USERNAME)
-            await self.async_set_unique_id(unique_id)
+            username = user_input.get(CONF_USERNAME)
+            password = user_input.get(CONF_PASSWORD)
+
+            await self.async_set_unique_id(username)
             self._abort_if_unique_id_configured()
```

Comment:
Where it come from?

Reply:
@tetienne what do you mean? `user_input` was already there. Not is it just more structured with  both retrieval of `username` and `password` at the top of this function.